notepad.py

```diff
@@ -56,11 +58,17 @@ def handleCommand(message, command, uID):
         if len(words) < 2:
             message.room.send_message('Missing duration argument.')
             return
-        try:
-            time = float(words[1])
-        except:
-            message.room.send_message('Number expected as first argument, got %s.'%words[1])
+
+        pattern = '(?:(?P<days>\d+)d)? \s* (?:(?P<hours>\d+)h)? \s* (?:(?P<minutes>\d+)m)? \s* (?:(?P<seconds>\d+)s)?'
+        res = re.match(pattern, words[1])
+        if not res:
+            message.room.send_message(words[1] + 'could not be parsed as duration.')
             return
+
+        spec = {key:int(val) if val else 0 for key,val in res.groupdict().items()}
```

Comment:
As I mentioned in a comment elsewhere, it might be better (more explicit) to use `key:int(val) if val is not None else 0` here.

Reply:
Isn't `None` falsy anyway? So this is just for clarity, right?

Reply:
`None` is falsey, but so would `''` be. But yes, we can't get `''` in a capture group because that doesn't match `'\d+'`. So yeah, I think the current version should function the same.